pkg/controller/iotconfig/auth_service.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2019, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package iotconfig
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/enmasseproject/enmasse/pkg/util/recon"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/enmasseproject/enmasse/pkg/util/install"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	iotv1alpha1 "github.com/enmasseproject/enmasse/pkg/apis/iot/v1alpha1"
+)
+
+const appAuthService = "iot-core"
+const nameAuthService = "iot-auth-service"
+
+func (r *ReconcileIoTConfig) processAuthService(ctx context.Context, config *iotv1alpha1.IoTConfig) (reconcile.Result, error) {
+
+	rc := &recon.ReconcileContext{}
+
+	rc.ProcessSimple(func() error {
+		return r.processDeployment(ctx, nameAuthService, config, r.reconcileAuthServiceDeployment)
+	})
+	rc.ProcessSimple(func() error {
+		return r.processService(ctx, nameAuthService, config, r.reconcileAuthServiceService)
+	})
+	rc.ProcessSimple(func() error {
+		return r.processConfigMap(ctx, nameAuthService+"-config", config, r.reconcileAuthServiceConfigMap)
+	})
+
+	return rc.Result()
+}
+
+func (r *ReconcileIoTConfig) reconcileAuthServiceDeployment(config *iotv1alpha1.IoTConfig, deployment *appsv1.Deployment) error {
+
+	install.ApplyDeploymentDefaults(deployment, "iot", appAuthService, deployment.Name)
```

Comment:
It would be nice to keep app: enmasse for all components, this makes it easier to find out which components are deployed by enmasse when you're looking at the whole cluster. Maybe you can add another label to be able to filter 'iot' components?

Reply:
the problem with that is, that "app" has a special meaning in the web console UI, and so you can group apps, by their different functionality. So instead of having one big blob of services, you have them grouped.

Reply:
In this case though, I think it's more important to adhere to existing conventions, i.e. tag all enmasse resources with 'app', because it has implications for those expecting this label to exist.

Reply:
> FWIW, if you want to influence UI on OpenShift 3.x, you can set annotations on the service to provide some level of grouping.

Do you have a pointer on this?

Reply:
*digs up enmasse 0.15.3*:

https://github.com/EnMasseProject/enmasse/blob/release-0.15.3/templates/include/messaging-service.jsonnet#L78

https://docs.openshift.com/container-platform/3.5/dev_guide/templates.html#other-recommendations

Reply:
Dependencies is something different however.

Reply:
Yeah I was hoping there was some additional annotations you could use. In any case I think its more important to make `oc`/`kubectl` experience as good as possible, and be able to query for all enmasse components (which is a superset of the 'iot' group). 

I wouldn't worry to much about the UI anyway since it will change.

Reply:
This is how it looks in the console (single app):

![image](https://user-images.githubusercontent.com/202474/54114385-999b2280-43ea-11e9-928f-c698b3604475.png)

Multiple apps:

![image](https://user-images.githubusercontent.com/202474/54114387-9d2ea980-43ea-11e9-913c-2835cb54d1d5.png)

Reply:
Done